nexus-repository-p2/src/main/java/org/sonatype/nexus/repository/p2/internal/metadata/CompositeRepositoryRewriter.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.p2.internal.metadata;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * Rewrites a p2 composite site collecting child site references and replacing them with site hashes.
+ *
+ * @since 1.next
```

Comment:
Are these maintained in community repos?

Reply:
This is a good Q.  I vaguely recall seeing some hard references.  There's nothing in the release process unless it's automated and I didn't notice.  @aornatovskyy you have an idea?

Reply:
I don't believe there is automation here, but there have been multiple releases since I opened this PR

Reply:
Right, but in case not clear, it's not clear to me whois responsible for updating 1.next.  If the answer is "no-one" it seems better to put a version?  If there's an update since you opened you would be updating.
See https://github.com/sonatype-nexus-community/nexus-repository-p2/blob/master/nexus-restore-p2/src/main/java/org/sonatype/nexus/blobstore/restore/p2/internal/P2RestoreBlobData.java for example, still says 0.next :(

Reply:
> Right, but in case not clear, it's not clear to me whois responsible for updating 1.next. If the answer is "no-one" it seems better to put a version? If there's an update since you opened you would be updating.

The plan is to replace them at merge time rather than end up with other incorrect versions.

Reply:
Ah gotcha, thanks for explain.

Reply:
@joedragons  sorry for the long response. It is not automated yet. Not sure we want to have it.